lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -1465,6 +1461,80 @@ def test_message_for_audit_mode(self):
+            invalidated_by=self.user
+        )
+        # Invalidate user certificate
+        generated_certificate.invalidate()
+        response = views._get_cert_data(self.user, self.course, self.course.id, True, CourseMode.HONOR)
+        self.assertEqual(response['cert_status'], 'invalidated')
+        self.assertEqual(response['title'], 'Your certificate has been invalidated')
+
+    def test_downloadable_get_cert_data(self):
+        """
+        Verify that downloadable cert data is returned if cert is downloadable.
+        """
+        self.generate_certificate(
+            "http://www.example.com/certificate.pdf", "honor"
+        )
+        with patch('certificates.api.certificate_downloadable_status', Mock(
+                return_value=self.mock_certificate_downloadable_status(is_downloadable=True))):
+            response = views._get_cert_data(self.user, self.course, self.course.id, True, CourseMode.HONOR)
+
+        self.assertEqual(response['cert_status'], 'downloadable')
+        self.assertEqual(response['title'], 'Your certificate is available')
+
+    def test_generating_get_cert_data(self):
+        """
+        Verify that generating cert data is returned if cert is generating.
+        """
+        self.generate_certificate(
+            "http://www.example.com/certificate.pdf", "honor"
+        )
+        with patch('certificates.api.certificate_downloadable_status',
+                   Mock(return_value=self.mock_certificate_downloadable_status(is_generating=True))):
+            response = views._get_cert_data(self.user, self.course, self.course.id, True, CourseMode.HONOR)
+
+        self.assertEqual(response['cert_status'], 'generating')
+        self.assertEqual(response['title'], "We're working on it...")
+
+    def test_unverified_get_cert_data(self):
+        """
+        Verify that unverified cert data is returned if cert is unverified.
+        """
+        self.generate_certificate(
+            "http://www.example.com/certificate.pdf", "honor"
+        )
+        with patch('certificates.api.certificate_downloadable_status',
+                   Mock(return_value=self.mock_certificate_downloadable_status(is_unverified=True))):
+            response = views._get_cert_data(self.user, self.course, self.course.id, True, CourseMode.HONOR)
+
+        self.assertEqual(response['cert_status'], 'unverified')
+        self.assertEqual(response['title'], "Certificate unavailable")
+
+    def test_request_get_cert_data(self):
+        """
+        Verify that requested cert data is returned if cert is to be requested.
+        """
+        self.generate_certificate(
+            "http://www.example.com/certificate.pdf", "honor"
+        )
+        with patch('certificates.api.certificate_downloadable_status',
+                   Mock(return_value=self.mock_certificate_downloadable_status())):
+            response = views._get_cert_data(self.user, self.course, self.course.id, True, CourseMode.HONOR)
```

Comment:
I take the above comment back. I now see you are unit testing the helper `_get_cert_data` function. Please rename the object from `response` to `cert_data` to avoid such confusion